src/Service/S3/src/S3Client.php

```diff
@@ -47,9 +49,27 @@
 use AsyncAws\S3\ValueObject\CommonPrefix;
 use AsyncAws\S3\ValueObject\MultipartUpload;
 use AsyncAws\S3\ValueObject\Part;
+use Psr\Log\LoggerInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 class S3Client extends AbstractApi
 {
+    /**
+     * @param S3Configuration|Configuration|array $configuration
+     */
+    public function __construct(
+        $configuration = [],
+        ?CredentialProvider $credentialProvider = null,
+        ?HttpClientInterface $httpClient = null,
+        ?LoggerInterface $logger = null
+    ) {
+        if (\is_array($configuration)) {
```

Comment:
But what if I use a configuration object. Then I have to choose between S3Configuration and Configuration.

Reply:
it's not an issue: If user provide a configuration, then it just don't benefit from S3 options.
The `instanceof` is check when client need to fetch "specific S3 configuration".

anyway this code is reverted